### DISCUSSION TITLE
How to create a nested one to many entity tree?

### DISCUSSION THREAD

**Discussion by pavelbelyaev98** [mikro-orm/mikro-orm#3608] [Q&A] [answered]
Hello, I am trying to created a nested tree of the same entity. 

In my case I need to create a  "tasks" table where each task can have multiple sub tasks where each sub task can have multiple subtasks and so on infinitely. (parent-children of the same entity)

Example:
```js
@Entity({ tableName: 'tasks' })
export class TaskEntity {
  @PrimaryKey()
  id: number;

  @Property()
  name: string;

  @OneToMany(() => TaskEntity, (task) => task.id)
  sub_tasks = new Collection<TaskEntity>(this);

  constructor(name: string) {
    this.name = name;
  }
}
```

When I am trying to create an entity Iike that I get the following error:
```TaskEntity.sub_tasks has wrong 'mappedBy' reference type: number instead of TaskEntity```

**Comment by B4nan** [comment, marked as answer]
>  `@OneToMany(() => TaskEntity, (task) => task.id)`

This is wrong, you need to define a m:1 property, you cant define only 1:m - it does not make sense, you need to have a FK that points to your parent entity, here you dont have one.

```ts
@Entity()
export class TaskEntity {
  @PrimaryKey()
  id: number;

  @Property()
  name: string;

  @ManyToOne(() => TaskEntity)
  parent?: TaskEntity;

  @OneToMany(() => TaskEntity, (task) => task.parent)
  children = new Collection<TaskEntity>(this);
}
```

**Reply by pavelbelyaev98** [reply]
I see, that makes sense, thanks!
